### PR TITLE
Fixing the rig-system

### PR DESCRIPTION
**Pull request by MarsM0nd**
Todo-List:
- [x] mounted gun shooting
- [x] magboots using the correct sprites when toggled
- [x] magboots are off while the suit isn't sealed
- [x] magboots disable if the suit has no power
- [x] \(Optional) magboots stay disabled if the suit has no power, as current magboots don't use power, i would remove/disable the disabling on powerdown. Disabling is commented out
- [x] headlamps act like magboots with forced disabling
- [x] ~~species specific sprites getting used~~ Species specific sprites worked, just not Tajaran.
- [x] Fixing Tajaran sprites getting used
- [x] Refurbishing current racial Spacesuit sprites for the rigs that are missing them. Done for Tajaran, Unathi, Skrell. Other races can be added at a later time
- [x] headlamps
- [x] Removal of a suit from someone via surgery to unseal, then normal stripping
- [x] Ensures rigparts can't be removed, need to be retracted to the backpart.
- [x] Rig modules don't loose/place the item they use somewhere
- [x] finding and fixing more bugs. List did grow a good bit, more might be found by playtesting.

### CODE REVIEW

File: code/modules/clothing/spacesuits/rig/rig.dm [hunk 1]
```diff
@@ -126,9 +126,11 @@
 	if(helm_type)
 		helmet = new helm_type(src)
 		verbs |= /obj/item/weapon/rig/proc/toggle_helmet
+		helmet.item_color="[initial(icon_state)]_sealed" //For the lightswitching to know the correct string to manipulate
 	if(boot_type)
 		boots = new boot_type(src)
 		verbs |= /obj/item/weapon/rig/proc/toggle_boots
+		boots.magboot_state="[initial(icon_state)]_sealed" //For the magboot (de)activation to know the correct string to manipulate
```
Comment (FlattestGuitar) [inline code/modules/clothing/spacesuits/rig/rig.dm:133]: Can you explain the point of this when [this](https://github.com/StratusSS13/Stratus/pull/21/files#diff-5597b35f1c3df45115d4e0dae367a094R146) exists? The linked line sets the icon state to just the initial one, which is what this line worked off of. This line doesn't seem to make a difference in the end.

File: code/modules/surgery/rig_removal.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
+//Procedures in this file: Unsealing a Rig.
+
+datum/surgery/rigsuit
```
Comment (FlattestGuitar) [inline code/modules/surgery/rig_removal.dm]: Missing a `/`. Literally the worst code ever.
👀

File: code/modules/surgery/rig_removal.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  name = "Rig Unsealing"
+  steps = list(/datum/surgery_step/rigsuit)
+  possible_locs = list("chest")
+
+datum/surgery/rigsuit/can_start(mob/user, mob/living/carbon/target)
```
Comment (FlattestGuitar) [inline code/modules/surgery/rig_removal.dm]: :eyes:
`/`

File: code/modules/surgery/rig_removal.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  if(istype(target,/mob/living/carbon/human))
```
Comment (FlattestGuitar) [inline code/modules/surgery/rig_removal.dm]: `if(ishuman(target))`

File: code/modules/clothing/spacesuits/rig/rig.dm (hunk 1 above)
Comment (MarsM0nd) [reply code/modules/clothing/spacesuits/rig/rig.dm:133]: `magboot_state` is a var of magboots, which use this to toggle between the two icons for on and off, `by icon_state = "[magboot_state][magpulse]"`

**Comment by MarsM0nd** [comment]
Travis doesn't seem to want to work, i'll close and reopen it.

**Pull request merged**